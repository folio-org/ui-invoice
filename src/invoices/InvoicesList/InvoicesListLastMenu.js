import React from 'react';
import { FormattedMessage } from 'react-intl';
import { withRouter } from 'react-router-dom';
import ReactRouterPropTypes from 'react-router-prop-types';

import {
  PaneMenu,
  Button,
} from '@folio/stripes/components';
import { IfPermission } from '@folio/stripes/core';

const InvoicesListLastMenu = ({ location }) => {
  return (
    <IfPermission perm="invoice.invoices.item.post">
      <PaneMenu>
        <FormattedMessage id="stripes-smart-components.addNew">
          {ariaLabel => (
            <Button
              id="clickable-newInvoice"
              aria-label={ariaLabel}
              to={{
                pathname: '/invoice/create',
                search: location.search,
              }}
              buttonStyle="primary"
              marginBottom0
            >
              <FormattedMessage id="stripes-smart-components.new" />
            </Button>
          )}
        </FormattedMessage>
      </PaneMenu>
    </IfPermission>
  );
};

InvoicesListLastMenu.propTypes = {
  location: ReactRouterPropTypes.location.isRequired,
};

export default withRouter(InvoicesListLastMenu);
