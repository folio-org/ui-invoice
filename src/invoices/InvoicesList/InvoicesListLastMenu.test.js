import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import '../../../test/jest/__mock__';

import { location } from '../../../test/jest/fixtures';

import { InvoicesListLastMenuComponent } from './InvoicesListLastMenu';

const renderInvoicesListLastMenu = () => render(
  <InvoicesListLastMenuComponent location={location} />,
  { wrapper: MemoryRouter },
);

describe('OrganizationsListLastMenu', () => {
  beforeEach(() => {
    global.document.createRange = global.document.originalCreateRange;
  });

  afterEach(() => {
    global.document.createRange = global.document.mockCreateRange;
  });

  it('should render correct structure', async () => {
    const { asFragment } = renderInvoicesListLastMenu();

    expect(asFragment()).toMatchSnapshot();
  });
});
