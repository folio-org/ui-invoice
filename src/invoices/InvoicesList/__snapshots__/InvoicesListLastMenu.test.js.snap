// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganizationsListLastMenu should render correct structure 1`] = `
<DocumentFragment>
  <section
    aria-labelledby="invoice-list-actions-label"
    class="menuSection"
    data-test-menu-section="true"
    id="invoice-list-actions"
  >
    <a
      aria-label="stripes-smart-components.addNew"
      class="button dropdownItem"
      href="/invoice/create?search"
      id="clickable-new-invoice"
      role="button"
    >
      <span>
        Icon
      </span>
    </a>
    <a
      aria-label="ui-invoice.button.voucherExport"
      class="button dropdownItem"
      href="/invoice/voucher-export?search"
      id="clickable-voucher-export"
      role="button"
    >
      <span>
        Icon
      </span>
    </a>
  </section>
</DocumentFragment>
`;
